### PR TITLE
180792001 refactor record button

### PR DESCRIPTION
**Pull request by Concord-Jim**
Greatly refactors the UX (and a good amount of code) for the user sound recording feature. Also hides the components/code which implemented the feature to display the selected sound's amplitude and wavelength. Finally, also prompts user to pick a sound (no sound is selected by default). 

PT story: https://www.pivotaltracker.com/story/show/180792001

Story details:
- move Record command into a button with the microphone logo.
- Pressing Record brings up a dialog that says "Press OK to begin recording for 5 seconds."
- Press OK, record for 5 seconds, and switch the sound dropdown to "My Recording"
- Record button is disabled during playback.
- remove labeling button/feature entirely.

Demo video: 
https://user-images.githubusercontent.com/91078832/149231297-5fc5ce34-28dc-491d-958c-8646a0ef95cb.mov

### CODE REVIEW

File: src/components/app.tsx
Comment (pjanik) [inline src/components/app.tsx:116]: It's very rare we really want to ignore this rule. So, I'd either put a comment that explains why it happened, or actually try to follow ESLint suggestions and update the array (which sometimes might lead to other changes, but in my personal experience - ESLint was always right :wink:). It seems to me that I'd rather do the latter in this case.

**Comment by mklewandowski** [comment]
I added @pjanik as a reviewer as I'm not familiar with this project.  I'll try to take a look nonetheless and see if I can chime in.

File: src/components/sound-picker/sound-picker.scss
Comment (mklewandowski) [inline src/components/sound-picker/sound-picker.scss:82]: Add newline

**Review by pjanik — CHANGES_REQUESTED** [review]
Looks good to me, thanks. 👍 

The only small change requested is not to ignore ESLint rules of hooks (or put a comment on why it happened). In my experience, it was never wrong, so I'm pretty cautious here.

Also, personally, I think I'm not a big fan of commenting out larger chunks of code. After a month no one will know if it's still useful or not, and the repo starts to look messy (the broken window effect). Also, when it's commented out, it's very likely things and types will diverge and it won't be easy to restore the feature later. So, if we think that there's value in this code and it might be restored one day, I wouldn't comment on most of it. I'd just hide or comment out the button and leave the rest untouched. Then, we'll at least get static type checking from TS and there are bigger chances it's possible to go back to this feature one day. But this is pretty opinionated, so I'd leave it up to you. Just my two cents.

File: src/components/app.tsx
Comment (mklewandowski) [inline src/components/app.tsx:231]: This indentation confused me.  Maybe easier to read as:
```
          <div
            className={`play-pause button${(selectedSound === "pick-sound") ? " disabled" : ""}`}
            onClick={handlePlay}
		  >
		    { playing
			  ? <PauseIcon className="pause-play-icons" />
			  : <PlayIcon className="pause-play-icons" />
		    }
          </div>
         ```

File: src/components/sound-picker/sound-picker.scss
Comment (mklewandowski) [inline src/components/sound-picker/sound-picker.scss:42]: I like these Sass variables to imply a color in the name (disabledButtonColor, disabledButtonGray, etc.) for clarity, but it likely doesn't matter at this point.

File: src/components/sound-picker/sound-picker.tsx
Comment (mklewandowski) [inline src/components/sound-picker/sound-picker.tsx:186]: check indentation on the if statement above.

File: src/hooks/use-sound-wave-rendering.ts
Comment (mklewandowski) [inline src/hooks/use-sound-wave-rendering.ts:224]: This is a lot of commented code that we are leaving in.

**Review by mklewandowski — CHANGES_REQUESTED** [review]
Looking good from what I can tell.  I made some comments for minor formatting changes.  My only other concern is the amount of commented code that is left in.  Is this intentional?  I was under the impression that we aren't coming back to this code any time soon so is it better to remove or even leave some of that code in place and uncommented?  Not sure exactly what the plan it so I may simply be out of the loop on what will happen to this code moving forward.

File: src/hooks/use-sound-wave-rendering.ts
Comment (Concord-Jim) [reply src/hooks/use-sound-wave-rendering.ts]: Just FYI, the Story Owner recommended commenting-out the code, over deleting it. But, in deference to team consensus, I've deleted it now.

Comment (kswenson) [reply src/hooks/use-sound-wave-rendering.ts]: If there's a realistic chance of wanting to return to it, the other approach would be to leave it in but behind a runtime flag -- which I think Piotr also suggested as a possible alternative. That way it's guaranteed that the code actually compiles and it's easy to come back to it if necessary. That's what I did with the circles/symbols at the end of the plot traces in Juice, for instance.

File: src/components/app.tsx
Comment (kswenson) [reply src/components/app.tsx:114]: >// Not including setupAudioContext() in dependency array; to avoid render loop.

If that's the reason, then the right solution is to wrap the `setupAudio()` function in `useCallback()` so that it's stable across renders.

File: src/components/app.tsx
Comment (kswenson) [inline src/components/app.tsx:8]: Looks like `isPureTone` is unused.

File: src/components/app.tsx
Comment (pjanik) [reply src/components/app.tsx:114]: Right, I'd vote to remove `// eslint-disable-next-line react-hooks/exhaustive-deps` and use `useCallback`. It seems like nitpicking, but I just think it's a good practice and bugs related to `useEffect` can be awful to trace (not necessarily in this case, but generally). Personally, I've only seen one (half) valid reason to skip this ESLint check since I started using `useEffect`. But it was actually the opposite - I've added extra variables to the dependency array to trigger update of non-react code from it. I've never had to remove a suggested dependency from this array.

File: src/components/zoom-buttons/zoom-buttons.tsx
Comment (mklewandowski) [inline src/components/zoom-buttons/zoom-buttons.tsx:30]: I think you mean for this closing bracket to be back one tab, but that's beyond nitpicking at this point

File: src/components/zoom-buttons/zoom-buttons.tsx
Comment (mklewandowski) [inline src/components/zoom-buttons/zoom-buttons.tsx:20]: I think you mean for this closing bracket to be back one tab, but that's beyond nitpicking at this point

**Review by mklewandowski — APPROVED** [review]
👍 I only went through quickly, but this looks good to me!  Best to let @pjanik take a look before merging since I'm not as familiar with this repo.

File: src/components/app.tsx
Comment (kswenson) [inline src/components/app.tsx:121]: The indirection through the module-level `setupAudioContext()` function is unnecessary.

**Review by kswenson — APPROVED** [review]
👍

**Review by pjanik — APPROVED** [review]

**Pull request merged**
